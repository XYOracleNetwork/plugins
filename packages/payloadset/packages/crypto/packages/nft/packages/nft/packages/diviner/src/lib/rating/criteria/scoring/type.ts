import { NftInfoFields } from '@xyo-network/crypto-nft-payload-plugin'
import { FAIL, PASS, PassFailScoringFunction } from '@xyo-network/crypto-nft-score-model'

export const scoreType: PassFailScoringFunction<NftInfoFields> = (nft: NftInfoFields) => {
  if (!nft.type) return FAIL
  if (typeof nft.type !== 'string') return FAIL
  const type = nft.type.toUpperCase()
  if (type !== 'ERC721' && type !== 'ERC1155') return FAIL
  return PASS
}
