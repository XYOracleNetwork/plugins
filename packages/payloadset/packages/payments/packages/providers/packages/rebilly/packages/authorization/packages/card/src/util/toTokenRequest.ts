import { BillingAddress, PaymentCard } from '@xyo-network/payment-payload-plugins'

import { CreatePaymentTokenRequest } from '../Api/index.ts'

/**
 * Converts a payment card and billing address to a token request
 * @param paymentCard The payment card
 * @param billingAddress The billing address
 * @returns The token request
 */
export const toTokenRequest = (paymentCard: PaymentCard, billingAddress: BillingAddress): CreatePaymentTokenRequest => {
  const {
    cvv, expMonth, expYear, cardNumber,
  } = paymentCard
  // NOTE: We are destructuring all but the schema and passing it along to the billingAddress. This
  // is to allow for missing fields to be omitted rather than set to undefined and for any new fields
  // that might be added later to be automatically picked up on. It is fine to pass this along since
  // the user is supplying this data about themselves.
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const { schema: _, ...fields } = billingAddress
  return {
    billingAddress: { ...fields },
    method: 'payment-card',
    paymentInstrument: {
      cvv, expMonth, expYear, pan: cardNumber,
    },
  }
}
