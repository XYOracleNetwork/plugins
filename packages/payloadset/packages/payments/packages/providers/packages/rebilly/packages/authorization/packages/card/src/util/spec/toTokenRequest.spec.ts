import type { BillingAddress, PaymentCard } from '@xyo-network/payment-payload-plugins'
import { BillingAddressSchema, PaymentCardSchema } from '@xyo-network/payment-payload-plugins'

import { toTokenRequest } from '../toTokenRequest'

describe('toTokenRequest', () => {
  const cases: [PaymentCard, BillingAddress][] = [
    [
      {
        cardNumber: '4111111111111111',
        cvv: '123',
        expMonth: 1,
        expYear: 3030,
        schema: PaymentCardSchema,
      },
      {
        // https://simpsons.fandom.com/wiki/123_Fake_Street
        address: '123 Fake Street',
        city: 'Oak Lawn',
        country: 'US',
        firstName: 'Marge',
        lastName: 'Simpson',
        postalCode: '60453',
        region: 'IL',
        schema: BillingAddressSchema,
      },
    ],
  ]
  it.each(cases)('convert a payment card and billing address to a token request', (paymentCard, billingAddress) => {
    const result = toTokenRequest(paymentCard, billingAddress)
    expect(result).toMatchSnapshot()
  })
})
