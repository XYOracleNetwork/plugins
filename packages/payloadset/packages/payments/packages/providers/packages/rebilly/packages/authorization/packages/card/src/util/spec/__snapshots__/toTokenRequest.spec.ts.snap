// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`toTokenRequest > convert a payment card and billing address to a token request 1`] = `
{
  "billingAddress": {
    "address": "123 Fake Street",
    "city": "Oak Lawn",
    "country": "US",
    "firstName": "Marge",
    "lastName": "Simpson",
    "postalCode": "60453",
    "region": "IL",
  },
  "method": "payment-card",
  "paymentInstrument": {
    "cvv": "123",
    "expMonth": 1,
    "expYear": 3030,
    "pan": "4111111111111111",
  },
}
`;

exports[`toTokenRequest convert a payment card and billing address to a token request 1`] = `
{
  "billingAddress": {
    "address": "123 Fake Street",
    "city": "Oak Lawn",
    "country": "US",
    "firstName": "Marge",
    "lastName": "Simpson",
    "postalCode": "60453",
    "region": "IL",
  },
  "method": "payment-card",
  "paymentInstrument": {
    "cvv": "123",
    "expMonth": 1,
    "expYear": 3030,
    "pan": "4111111111111111",
  },
}
`;
